### PR TITLE
adds basic refund/capture endpoints

### PR DESCRIPTION
**Pull request by pi3r**
adds basic refund/capture endpoints.

/cc @andrewpaliga @jordanliddle

### CODE REVIEW

**Comment by jordanliddle** [comment]
Amazing - the ability to refund from admin for HPSDK integrations has been requested over and over again. This will have such a positive impact!

I attempted to 🎩  refund ability with my existing gateway (id 1030450), but it keeps failing on signature validation [here](https://github.com/Shopify/hosted-payment-sim/blob/281c8d8cd9bc7ab279931d0cdeb541cf9da8907b/app.rb#L37). I'm probably doing something wrong here, but could it possible that the my gateway credential2 is not being pulled [as expected  when being signed](https://github.com/Shopify/shopify/blob/master/app/services/payments/offsite_refund_service.rb#L82)? 

[example order](https://liddle.myshopify.com/admin/orders/3997081349)
@Busfox could you do a sanity check on this and 🎩 on your shop?

**Comment by jordanliddle** [comment]
@ShayneP for 🎩 - lemme know if you need a hand setting up the environment

**Comment by jordanliddle** [comment]
With the new changes in #92171, this now works as expected. "Refund created successfully!" 

🚢 :shipit:

File: app.rb [hunk 1]
```diff
@@ -15,7 +15,11 @@ def initialize(base_path: '')
   end
 
   def fields
-    @fields ||= request.params.select {|k, v| k.start_with? 'x_'}
+    @fields ||= if request.content_type == 'application/json'
+      JSON.load(request.body.read)
+    else
+      request.params.select {|k, v| k.start_with? 'x_'}
```
Comment (Krystosterone) [inline app.rb]: Nitpick but this is hard to read. Could you change it to `.select { |k, v| k.start_with?('x_') }`

**Comment by Krystosterone** [comment]
Two small things:
1. I think there was still one 👍  missing
2. I'm surprised no tests changed

**Comment by pi3r** [comment]
@Krystosterone oh 💩 , I thought you :+1:'d this one... my bad. Well there's no test on the app but I can add them.

**Comment by dinesh** [comment]
@pi3r is there any documentation for capture and refund endpoints and what parameters are expected in request/response ? how to configure these endpoints in shopify shop ?

**Comment by pi3r** [comment]
@dinesh not yet, this is still early stages. We will update the documentation once it's ready.

**Pull request merged**